ps_socialfollow.php

```diff
@@ -283,13 +320,27 @@ protected function updateFields()
     {
         $validator = Validation::createValidator();
         $constraints = [new Url()];
-
+        $config = [];
+        $errors = [];
         foreach (static::SOCIAL_NETWORKS as $social) {
-            $value = Tools::getValue('blocksocial_' . $social, '');
-            $violations = $validator->validate($value, $constraints);
-            if (0 === count($violations)) {
-                Configuration::updateValue('BLOCKSOCIAL_' . strtoupper($social), $value);
+            foreach ($this->context->controller->getLanguages() as $lang) {
+                $id_lang = $lang['id_lang'];
+                $config[$social][$id_lang] = Tools::getValue('blocksocial_' . $social . '_' . $id_lang, '');
+                $violations = $validator->validate($config[$social][$id_lang], $constraints);
+                if (count($violations)) {
```

Comment:
Nice-to-have: insert a blank line above the `if` to ease reading

Reply:
OK